doc/source/custom_directives.py

```diff
@@ -776,8 +776,8 @@ def render_library_examples(config: pathlib.Path = None) -> bs4.BeautifulSoup:
         soup.append(page_text)
 
         container = soup.new_tag("div", attrs={"class": "example-index"})
-        for group, examples in examples.items():
```

Comment:
why this change?

Reply:
Because if we don't change examples to _example, it will trigger flake8's warning:

"Found for loop that reassigns the iterable it is iterating with each iterable value. Flake8 (B020)"

Reply:
By the way, regarding a PR that only changes type() == x to isinstance, do I need to close the current PR and open a new one for this?"

Reply:
My recommendation is that you create a PR for each flake8 violation code and then this PR will just be the last PR that upgrades flake8.

Reply:
OK, I'll work on it shortly. Thanks for your suggestion!

Reply:
@jjyao I agree with you. I'll help @CheyuWu divide this PR into several smaller ones offline, and possibly assign some tasks to other contributors.

Reply:
I think we should avoid using variable names that start with an underscore. We can discuss how to name these variables after I divide this task into smaller ones.